src/AsmResolver.Workspaces.DotNet/DotNetRelations.cs

```diff
@@ -47,13 +47,28 @@ public static class DotNetRelations
             "ReferenceMember",
             new Guid("ce11d2f6-a423-429d-ad37-2f073fdf63be"));
 
+        /// <summary>
+        /// Describes the relationship between a method definition and its specification.
+        /// </summary>
+        public static readonly ObjectRelation<MethodDefinition, MethodSpecification> ReferenceMethodSpecification = new(
+            "ReferenceMethodSpecification",
+            new Guid("3824566f-3df4-4188-8e67-ed2d745df84a"));
+
+
         /// <summary>
         /// Describes the relationship between a type definition and its reference.
         /// </summary>
         public static readonly ObjectRelation<TypeDefinition, TypeReference> ReferenceType = new(
             "ReferenceType",
             new Guid("3cc86779-338c-4165-a00c-da547a2e8549"));
 
+        /// <summary>
+        /// Describes the relationship between a type definition and its specification.
+        /// </summary>
+        public static readonly ObjectRelation<TypeDefinition, TypeSpecification> ReferenceTypeSpecification = new(
```

Comment:
I don't understand the purpose of this relation. `TypeSpecification`s assign tokens to `TypeSignature`s. TypeSignatures can contain references to other types (e.g. `GenericType<TypeParam1, TypeParam2>` contains 3 types). Furthermore, these types within the signatures can be definitions as well as references. Reducing it to just type definitions flattens the dependency graph, no?

Reply:
Furthermore, they are also used for other decorator type sigs like `int[][][]` or `int[]&`. How does this fit then in this relation?

Reply:
This relation is later used in pr #249 for linking/chaining base types when they are type specification
in src/AsmResolver.Workspaces.DotNet/Analyzers/Implementation/ImplementationUtilities.cs
internal static IEnumerable<ITypeDefOrRef> GetBaseTypes(WorkspaceIndexNode baseNode)

Reply:
Maybe it should be named differently, for example ReferenceBaseTypeSpecification  for chaining TypeSpecification and its TypeDefOrRef generated from signature.

Or to do this with each TypeSignature , but only the first one (skipping the nested ones) for less complexity

Reply:
Why not just `TypeDefinition` to `ITypeDefOrRef`, and reuse the previously defined relation?